app/src/main/java/com/lagradost/cloudstream3/MainActivity.kt

```diff
@@ -570,18 +568,37 @@ class MainActivity : AppCompatActivity(), ColorPickerDialogListener, BiometricCa
                 false
             }
         }
+
         binding?.apply {
-            navView.isVisible = isNavVisible && !landscape
             navRailView.isVisible = isNavVisible && landscape
+            navView.isVisible = isNavVisible && !landscape
 
-            // Hide library on TV since it is not supported yet :(
-            //val isTrueTv = isTrueTvSettings()
-            //navView.menu.findItem(R.id.navigation_library)?.isVisible = !isTrueTv
-            //navRailView.menu.findItem(R.id.navigation_library)?.isVisible = !isTrueTv
+            // Change library icon with logo of current api in sync
+            val libraryViewModel = ViewModelProvider(this@MainActivity)[LibraryViewModel::class.java]
+            val lastAPI = getKey<String>("$currentAccount/$LAST_SYNC_API_KEY").toString()
 
-            // Hide downloads on TV
-            //navView.menu.findItem(R.id.navigation_downloads)?.isVisible = !isTrueTv
-            //navRailView.menu.findItem(R.id.navigation_downloads)?.isVisible = !isTrueTv
+            fun setLibraryIcon(drawable: Int) {
+                navRailView.apply {
+                    menu.findItem(R.id.navigation_library)?.setIcon(drawable)
+                }
+                navView.apply {
+                    menu.findItem(R.id.navigation_library)?.setIcon(drawable)
+                }
+            }
+
+            fun changeLibraryIcon() {
+                when (lastAPI) {
```

Comment:
Why? You quite literally observe currentApiName so why not use that?

Reply:
because if we do that, then it requires the user to go to library everytime, then it starts observing it, thus everytime the icon will be resetted, and until LVM is not observed